beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -174,6 +174,9 @@ func (vs *Server) getPayloadHeaderFromBuilder(ctx context.Context, slot primitiv
 	defer cancel()
 
 	signedBid, err := vs.BlockBuilder.GetHeader(ctx, slot, bytesutil.ToBytes32(h.BlockHash()), pk)
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
```

Comment:
There is no need to check the context expired here, the block builder client would have returned an error if the context was expired.

Reply:
OK, you're actually right, I see now that the original call ends up with `http.NewRequestWithContext()`. I got mislead by the mocked `GetHeader()`...